### PR TITLE
Linked List Palindrome Check.cpp

### PR DESCRIPTION
**Pull request by AnushkaS09**
Check if the linked list is palindrome or not if yes return true if not return false. Linked list has to be taken as an input from the user.
ISSUE: #1340

### CODE REVIEW

**Comment by AnushkaS09** [comment]
@tarun26091999 @plazzy99 @rudrakshi99 Please review

File: Linked_list/PalindromeCheck.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ enter the linked list(enter -1 at the end of the linked list):3 1 3 -1
+ true
+ 1 2 3 -1
+ false
+ ----------------------------------------------------------------------------------------
+TIME COMPLEXITY: O(N)
+SPACE COMPLEXITY: O(1)
+*/
+
+
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:56]: remove extra lines

File: Linked_list/readme.md [hunk 2]
```diff
@@ -34,7 +34,6 @@ properties.
 * Splitting ----> [C++](/Code/C++/Splitting_Linked_lists.cpp)
 * Implementation of Linked List ----> [Python](/Code/Python/linked_list.py)
 * Reversing a Linked List ----> [C++](/Code/C++/reverse_a_linked_list.cpp)
+* Palindrome Check---->[C++](/Linked_list/PalindromeCheck.cpp)
```
Comment (rish-singhal) [inline Linked_list/readme.md:37]: sort this list in alphabetical order

File: Linked_list/PalindromeCheck.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    Node *new_head = reverse(save->next);
+    Node *temp = save->next;      
+    temp->next = save;
+    save->next = NULL;
+    return new_head; 
+}
+bool isPalindrome(Node *head)
+{
+    Node *help=head;
+  Node *temp = reverse(help);
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:19]: use proper indentation of 1 tab or 4 spaces!

File: Linked_list/PalindromeCheck.cpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
+/*Check if the linked list is palindrome or not if yes return true if not return false. Linked list has to be taken as an input from the user.*/
+Node* reverse(Node *save)
+{
+   if(save==NULL || save->next==NULL)  
+    {
+
+        return save;
+    }
+    
+    Node *new_head = reverse(save->next);
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:10]: add appropriate comments explaining your algorithm and what variable does

File: Linked_list/PalindromeCheck.cpp [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/*Check if the linked list is palindrome or not if yes return true if not return false. Linked list has to be taken as an input from the user.*/
+Node* reverse(Node *save)
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:2]: add spaces here

File: Linked_list/PalindromeCheck.cpp [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        else{
+            return 1;
+        }
+        temp = temp->next;
+        head = head->next;
+    }
+}
+/*
+----------------------------------------------------------------------------------
+SAMPLE TEST CASE 1:
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:37]: make the code interactive!

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by tarun26091999** [comment]
main function is not present

File: Linked_list/PalindromeCheck.cpp [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    
+           Node *new_head = reverse(save->next);//RECURSIVE CALL
+           Node *temp = save->next;      
+           temp->next = save;
+           save->next = NULL;
+           return new_head; 
+       }
+       bool isPalindrome (Node *head)
+       {
+           Node *help=head;
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:33]: add spaces around =

File: Linked_list/PalindromeCheck.cpp [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+           Node *temp = reverse(help);//reversing the linked list ans saving its head in temp
+    
+           if(head==NULL){
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:36]: add spaces around ==

File: Linked_list/PalindromeCheck.cpp [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+               return 1;
+           }
+           while(temp && help){
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:39]: add space after while(){ ===> i.e. while() {...

Do this with every if and while conditions in the code

File: Linked_list/PalindromeCheck.cpp [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+                   return 0;
+               }
+               else{
+                   return 1;
+               }
+               temp = temp->next;
+               head = head->next;
+           }
+       }
+Node *takeinput()
```
Comment (rish-singhal) [inline Linked_list/PalindromeCheck.cpp:50]: add an empty line here

File: Linked_list/readme.md (hunk 2 above)
Comment (rish-singhal) [reply Linked_list/readme.md:44]: sort complete list .

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Pull request merged**